src/kagglehub/models_helpers.py

```diff
@@ -9,18 +9,32 @@
 logger = logging.getLogger(__name__)
 
 
+class Directory:
+    name: str
+    files: List[str]
+    directories: List["Directory"]
+
+
+FileStructure = List[Directory]
+
+
 def _create_model(owner_slug: str, model_slug: str) -> None:
     data = {"ownerSlug": owner_slug, "slug": model_slug, "title": model_slug, "isPrivate": True}
     api_client = KaggleApiV1Client()
     api_client.post("/models/create/new", data)
     logger.info(f"Model '{model_slug}' Created.")
 
 
-def _create_model_instance(model_handle: ModelHandle, files: List[str], license_name: Optional[str] = None) -> None:
+def _create_model_instance(
+    model_handle: ModelHandle, files_and_directories: FileStructure, license_name: Optional[str] = None
+) -> None:
+    print([subdir for subdir in files_and_directories["directories"]])
+    print([{"token": file_token} for file_token in files_and_directories["files"]])
     data = {
         "instanceSlug": model_handle.variation,
         "framework": model_handle.framework,
-        "files": [{"token": file_token} for file_token in files],
+        "files": [{"token": file_token} for file_token in files_and_directories["files"]],
+        "directories": [subdir for subdir in files_and_directories["directories"]],
```

Comment:
you will need to serialize this to a map matching the proto request format. Similar to how you did this above for "files". You will need to do this recursively. I would recommend to add a method to serialize it to `UploadDirectoryInfo`. Same, you could add it to `UploadFileInfo` too.